backend/src/apiserver/server/api_converter.go

```diff
@@ -148,14 +152,48 @@ func toApiParameters(paramsString string) ([]*api.Parameter, error) {
 	return apiParams, nil
 }
 
+func toApiRuntimeConfig(modelRuntime model.RuntimeConfig) (*api.PipelineSpec_RuntimeConfig, error) {
+	var runtimeParamsStringsMap map[string]string
+	var runtimeParams map[string]*structpb.Value
+	if modelRuntime.Parameters != "" {
+		err := json.Unmarshal([]byte(modelRuntime.Parameters), &runtimeParamsStringsMap)
+		if err != nil {
+			return nil, util.NewInternalServerError(err, fmt.Sprintf("Cannot unmarshal RuntimeConfig Parameter to map[string]string, string value: %+v", modelRuntime.Parameters))
+		}
+		runtimeParams = make(map[string]*structpb.Value)
+		for k, v := range runtimeParamsStringsMap {
+			var protoValue structpb.Value
+			err := protoValue.UnmarshalJSON([]byte(v))
+			if err != nil {
+				return nil, util.NewInternalServerError(err, fmt.Sprintf("Cannot unmarshal string %+v to structpb.Value", v))
+			}
+			runtimeParams[k] = &protoValue
+		}
+	}
+	apiRuntimeConfig := &api.PipelineSpec_RuntimeConfig{
+		Parameters:   runtimeParams,
+		PipelineRoot: modelRuntime.PipelineRoot,
+	}
+	return apiRuntimeConfig, nil
+}
+
 func toApiRun(run *model.Run) *api.Run {
+	// v1 parameters
 	params, err := toApiParameters(run.Parameters)
 	if err != nil {
 		return &api.Run{
 			Id:    run.UUID,
 			Error: err.Error(),
 		}
 	}
+	// v2 RuntimeConfig
+	runtimeConfig, err := toApiRuntimeConfig(run.PipelineSpec.RuntimeConfig)
+	if err != nil {
+		return &api.Run{
+			Id:    run.UUID,
+			Error: err.Error(),
+		}
+	}
```

Comment:
I'm still feeling a bit strong about this. 
we should really create a new v2 pipeline_spec.proto instead of reusing v1 API in
https://github.com/kubeflow/pipelines/blob/master/backend/api/pipeline_spec.proto

To note, anything we released to post Beta is subject to the formal deprecation policy. If I understand correctly this will be released as part of the next V1 release which is already GA'ed? (Or is the API proto already released?)

Reply:
that being said i'm ok with merge this and do a quick follow up PR to properly break down the code before the next backward compatible promised release.

Reply:
Yes, the current API that has been released (which is going to be V1 going forward) already contains `RuntimeConfig` field. FE uses `RuntimeConfig` to pass parameters and pipeline roots, where only parameters are stored in the backend DB. Then backend returns the run without using the `RuntimeConfig` field. The change here just makes sure the `RuntimeConfig` info coming from UI is stored, and then can be returned via `RuntimeConfig`.  Shall we remove `RuntimeConfig` from V1 entirely after releasing V2?

Reply:
Yes let's call out that in the proto comment, and do that when adding v2 api.